dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GithubAppCheck.java

```diff
@@ -75,40 +83,70 @@ public GithubAppCheck(final Namespace ns){
+
+
+    protected boolean isGithubAppEnabledOnRepositoryWithGitApi_Retry(String fullRepoName) {
+        RetryConfig config = RetryConfig.custom()
+                .maxAttempts(1)
+                .waitDuration(Duration.ofMillis(1000))
+                .retryExceptions(IOException.class)
+                .build();
+        Retry retry = Retry.of("id", config);
+
+        Try<Boolean> retryResult = Try.ofSupplier(Retry.decorateSupplier(retry, () -> isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName)));
+        
+        return retryResult.get();
     }
 
+
     /**
      * Method to verify whether the github app is installed on a repository or not, using Git API
      * @param fullRepoName = The repository full name, i.e, of the format "owner/repoName". Eg: "Salesforce/dockerfile-image-update"
      * @return True if github app is installed, false otherwise. 
      * Reference: https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-a-repository-installation-for-the-authenticated-app
      */
+
     protected boolean isGithubAppEnabledOnRepositoryWithGitApi(String fullRepoName) {
+        return isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName, httpClient);
+    }
+
+    protected boolean isGithubAppEnabledOnRepositoryWithGitApi(String fullRepoName, CloseableHttpClient httpClient) {
         refreshJwtIfNeeded(appId, privateKeyPath);
         try {
-            gitHub.getApp().getInstallationByRepository(fullRepoName.split("/")[0], fullRepoName.split("/")[1]);
-            return true;
-        } catch (HttpException exception) {
-            if (exception.getResponseCode() != 404) {
-                // Log for any HTTP status code other than 404 Not found. 
-                log.warn("Caught a HTTPException {} while trying to get app installation. Defaulting to False", exception.getMessage());
+            String apiEndpoint = "https://git.soma.salesforce.com/api/v3/repos/" + fullRepoName + "/installation";
+            HttpGet httpGet = new HttpGet(apiEndpoint);
+            httpGet.setHeader("Authorization", jwt);
+            httpGet.setHeader("Accept", "application/vnd.github+json");
+            try (CloseableHttpResponse response = httpClient.execute(httpGet)) {
+                int statusCode = response.getStatusLine().getStatusCode();
+                log.warn("[isGithubAppEnabledOnRepositoryWithGitApi] -- Response code `{}` while trying to get app installation using Git API", statusCode);
+                return statusCode == 200;
             }
-            return false;
         } catch (IOException exception) {
-            // Most often happens on timeout scenarios. 
-            log.warn("Caught a IOException {} while trying to get app installation. Defaulting to False", exception.getMessage());
+            log.warn("[isGithubAppEnabledOnRepositoryWithGitApi] -- Caught a IOException while trying to get app installation on repo {}. Defaulting to False", fullRepoName);
+            exception.printStackTrace();
```

Comment:
Include exception as a parameter to log.warn, the stack trace should be included in the log message.

Reply:
@cjneasbi resolved